src/server.h

```diff
@@ -1386,6 +1386,16 @@ typedef struct client {
         net_output_bytes_curr_cmd; /* Total network output bytes sent to this client, by the current command. */
 } client;
 
+typedef struct {
+    uint64_t id;
```

Comment:
A filter should be able to have a list of IDs, as in `CLIENT LIST ID 3 4 5`. We need some list here, or maybe an `intset`? (See intset.c)

We need some comments too. Does 0 and NULL for these fields mean match any client? It's worth mentioning in a comment.

This struct and the new functions can probably be just local in networking.c. No other files need to use these.